### DISCUSSION TITLE
redirection and sitemap.xml conflict?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#37652] [Help] [answered]
OK... I think screw something up...

I got the sitemap.xml at the root folder and I could view it on the browser going to mydomain.com/sitemap.xml

I also got a redirect working  from a 404 to the home page /

Then.... I went to vercel and changed my settings because I wanted to see www.mydomain.com instead of just mydomain.com 

So cool.... I added the www in front of the domain... but now I can't view the xml folder in the browser because  when I go to www.mydomain/sitemap.xml it directs me to home at www.mydomain.com 

Is there a way to fix that?

**Comment by icyJoseph** [comment]
The `sitemap.xml` must go inside `public`. I've corrected our misunderstanding in another discussion I believe. Not really sure how it worked out for you in the root though, it must always go inside `public` https://nextjs.org/docs/basic-features/static-file-serving

**Comment by Elindo586** [comment]
Ok... so I put the sitemap.xml in the public folder...   it goes like this:

https://www.mywebsite.com/public/sitemap.xml

but if I try that in the browser, or on the localhost:3000  it  redirects me back to the home page  as if the sitemap was a 404 page that doesn't exist.

How could I fix this?

**Reply by icyJoseph** [reply]
When you put it in the public folder, then it is just available at the root of the site.

1. Put the file in public, `pubic/sitemap.xml`
2. In the browser go to `example.com/sitemap.xml`

If it's not working try to reload the page with `ctrl+shift+r`

## Edit

Tested with your repo and it works as I describe.

**Comment by icyJoseph** [comment]
From the next.js docs:

> Next.js can serve static files, like images, under a folder called public in the root directory. Files inside public can then be referenced by your code starting from the base URL (/).

> For example, if you add an image to public/me.png, the following code will access the image:

```jsx
import Image from 'next/image'

function Avatar() {
  // no public, only /me.png
  return <Image src="/me.png" alt="me" width="64" height="64" />
}

export default Avatar
```

> This folder is also useful for robots.txt, favicon.ico, Google Site Verification, and any other static files (including .html)!

https://nextjs.org/docs/basic-features/static-file-serving

Please do read the docs carefully, these document all of these use cases.

If you are using `rewrites` make sure to use the `afterFiles` rewrites!

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return {
      afterFiles: [{ source: "/:path*", destination: "/_404/:path*" }],
    };
  },
};
```

**Comment by Elindo586** [comment, marked as answer]
[icyJoseph](https://github.com/icyJoseph)
[1 hour ago](https://github.com/vercel/next.js/discussions/37652#discussioncomment-2938375)
When you put it in the public folder, then it is just available at the root of the site.

Put the file in public, pubic/sitemap.xml
In the browser go to example.com/sitemap.xml
If it's not working try to reload the page with ctrl+shift+r

Edit
Tested with your repo and it works as I describe.
